app/public/components/chips/indexChip.js

```diff
@@ -16,27 +16,27 @@ import { h } from '/js/src/index.js';
 import { RCT } from '../../config.js';
 const { dataReqParams, pageNames } = RCT;
 
-export default function indexChip(model, index) {
+export default function indexChip(model, pageName, index) {
     const dataPointer = model.getCurrentDataPointer();
-    const { page } = dataPointer;
-    const data = model.fetchedData[dataPointer.page][dataPointer.index].payload;
+    const currentPage = dataPointer.page;
+    const data = model.fetchedData[currentPage][dataPointer.index].payload;
     const { fields } = data;
     const firstField = fields.find((f) => f !== undefined && f.name);
-    const targetUrl = `/?page=${page}&index=${index}&${dataReqParams.rowsOnSite}=50&${dataReqParams.site}=1&sorting=-${firstField.name}`;
+    const targetUrl = `/?page=${currentPage}&index=${index}&${dataReqParams.rowsOnSite}=50&${dataReqParams.site}=1&sorting=-${firstField.name}`;
 
-    return page !== pageNames.periods && model.fetchedData[page][index]
+    return currentPage !== pageNames.periods && model.fetchedData[currentPage][index] && currentPage === pageName
```

Comment:
Variables names of 'page' and 'pageName'  not sufficiently discriminative against their purpose